server/market/market.go

```diff
@@ -128,32 +135,64 @@ func (m *Market) SubmitOrderAsync(rec *orderRecord) <-chan error {
 	return sig.errChan
 }
 
+// MidGap returns the mid-gap market rate, which is ths rate halfway between the
+// best buy order and the best sell order in the order book.
+func (m *Market) MidGap() uint64 {
+	bestBuy, bestSell := m.book.Best()
+	return (bestBuy.Rate + bestSell.Rate) / 2
+}
+
+// CoinLocked checks if a coin is locked. The asset is specified since we should
+// not assume that a CoinID for one asset cannot be made to match another
+// asset's CoinID.
+func (m *Market) CoinLocked(asset uint32, coin coinlock.CoinID) bool {
```

Comment:
I made a comment in the order router too, but if it's easier to pass the `DEXCoinLocker` to the order router, this method could be dropped from `MarketTunnel`.